### ISSUE TITLE
False positives with keywords-only fields of dataclasses

### ISSUE THREAD

**Issue by NeilGirdhar** [PyCQA/pylint#5767]
### Bug description

```python
from dataclasses import dataclass, KW_ONLY

@dataclass
class C:
    _: KW_ONLY
    c: int

c = C(c=1)  #  No value for argument '_' in constructor call (no-value-for-parameter)
```
### Pylint version

```shell
pylint 2.12.2
astroid 2.9.3
Python 3.10.0 (default, Nov  6 2021, 07:41:13) [Clang 12.0.5 (clang-1205.0.22.9)]
```

**Comment by NeilGirdhar** [comment]
Related https://github.com/PyCQA/pylint/issues/3492

**Comment by Infernio** [comment]
Happens with the `kw_only` attribute of `dataclasses.field` too:
```python
"""Quick test to demonstrate pylint errors when using kw_only dataclass
fields."""
from dataclasses import dataclass, field

@dataclass
class Creature:
    """Defines a creature with one or more names (first names, middle names,
    last names, etc.)"""
    names: list[str] = field(kw_only=True)

@dataclass
class Human(Creature):
    """Defines a human who has a home address."""
    address: str

donna = Human('1858 Hidden Meadow Drive, East Grand Forks, ND 56721',
              names=['Donna', 'Jennings'])
print(donna)
```

Running pylint (2.12.2):
```
$ python -m pylint dc_kw_only.py 
************* Module dc_kw_only
dc_kw_only.py:16:8: E1124: Argument 'names' passed by position and keyword in constructor call (redundant-keyword-arg)
dc_kw_only.py:16:8: E1120: No value for argument 'address' in constructor call (no-value-for-parameter)

--------------------------------------------------------------------
Your code has been rated at -4.29/10
```

**Comment by jacobtylerwalls** [comment]
A valuable test case was given in #7290 that should pass when fixing this issue.

**Comment by DanielNoord** [comment]
I opened #7333 to fix the issue with `KW_ONLY`.

@Infernio I have decided to track your issue in #7333 as it is not actually related in terms of what needs to be done. It seems related but sadly isn't. I might be able to take a look though after we land #7333.
